editors/emacs/Portfile

```diff
@@ -101,19 +101,16 @@ if {$subport eq $name || $subport eq "emacs-app"} {
 }
 
 if {$subport eq "emacs-devel" || $subport eq "emacs-app-devel"} {
-    set date        2022-09-20
     epoch           5
-    version         [string map {- {}} ${date}]
+    version         20221029
+    set git_commit  ce527464440ca070e752ca18a68ebf2d369a14df
     revision        0
 
-    fetch.type      git
-    git.url         --shallow-since=${date}T00:00:00 https://github.com/emacs-mirror/emacs.git
-    git.branch      e5de29aa47e96f675b7f4ef3a7f84bd30d68ea0f
+    master_sites    https://github.com/emacs-mirror/emacs/archive/${git_commit}.tar.gz?dummy=
```

Comment:
My understanding is that fetching from git is desirable because we don't really want to waste mirror space on these ephemeral development versions. Is there a particular reason to switch to tarball?

Reply:
@amake to enjoy checksums! I may add `NoMirror` to exclude it. Or revert this changes.

BTW `-devel` port was introduced here: https://github.com/macports/macports-ports/commit/fb4f094976b6fe31f22263492f5348f8e1e380b5 and it was made with `fetch.type git` because git.savannah.gnu.org hasn't got a nice way to download an archive from it.

The switch to github was made here https://github.com/macports/macports-ports/commit/f62654eb19904b3bdc34336d94cb7f799a01ebe2

And I've introduced `--depth` to make things faster at https://github.com/macports/macports-ports/commit/68c42a1a1bff35daea513a761bdf9da6cf95192e

Which was converted to `--shallow-since` with dependency to git.

=> I haven't found any reason to not use github archive's to be honest.

Reply:
I have no problem pulling the archive from GitHub. My concern was first about mirroring, and second about the fact that the distfile will now sit on my machine unless I go to delete it (admittedly a small cost).

But if we are going to go this route, wouldn't it make more sense to either use `PortGroup github` or at least follow the same conventions? This is almost `github.tarball_from archives` but a little different. Why `archives` and not `tarball`?

Reply:
@catap You requested my review again; maybe you missed my questions above?

Reply:
@amake sorry, I've missed your question.

To be honest I like an idea with `github` PG and just rebased my PR to this idea. It makes things cleaner.